iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/CertificateBasedAuthenticationCheck.java

```diff
@@ -19,35 +19,69 @@
  */
 package org.sonar.iac.arm.checks;
 
+import java.util.LinkedList;
+import java.util.List;
+import org.sonar.iac.arm.tree.ArmTreeUtils;
 import org.sonar.iac.arm.tree.api.ArmTree;
 import org.sonar.iac.arm.tree.api.BooleanLiteral;
 import org.sonar.iac.arm.tree.api.ResourceDeclaration;
+import org.sonar.iac.arm.tree.api.StringLiteral;
 import org.sonar.iac.common.api.checks.CheckContext;
 import org.sonar.iac.common.api.tree.Tree;
 import org.sonar.iac.common.checks.PropertyUtils;
 
 public class CertificateBasedAuthenticationCheck extends AbstractArmResourceCheck {
 
-  private static final String MISSING_CERTIFICATE = "Omitting \"%s\" disables certificate-based authentication. Make sure it is safe here.";
-  private static final String DISABLED_CERTIFICATE = "Make sure that disabling certificate-based authentication is safe here.";
+  private static final String MISSING_CERTIFICATE_MESSAGE = "Omitting \"%s\" disables certificate-based authentication. Make sure it is safe here.";
+  private static final String DISABLED_CERTIFICATE_MESSAGE = "Make sure that disabling certificate-based authentication is safe here.";
+  private static final String ALLOWING_NO_CERTIFICATE_MESSAGE = "Connections without client certificates will be permitted. Make sure it is safe here.";
+
+  private static final List<String> PATH_CONTAINER_APPS = ArmTreeUtils.computePath("configuration/ingress");
 
   @Override
   protected void registerResourceConsumer() {
-    register("Microsoft.ApiManagement/service/gateways/hostnameConfigurations", CertificateBasedAuthenticationCheck::checkNegotiateClientCertificate);
+    register("Microsoft.ApiManagement/service/gateways/hostnameConfigurations", CertificateBasedAuthenticationCheck::checkHostnameConfigurations);
+    register("Microsoft.App/containerApps", CertificateBasedAuthenticationCheck::checkContainerApps);
   }
 
-  private static void checkNegotiateClientCertificate(CheckContext ctx, ResourceDeclaration resource) {
-    PropertyUtils.value(resource, "negotiateClientCertificate")
+  private static void checkHostnameConfigurations(CheckContext ctx, ResourceDeclaration resource) {
+    PropertyUtils.get(resource, "negotiateClientCertificate")
       .ifPresentOrElse(
-        value -> {
-          if (isBooleanFalse(value)) {
-            ctx.reportIssue(value, DISABLED_CERTIFICATE);
+        property -> {
+          if (isBooleanFalse(property.value())) {
+            ctx.reportIssue(property, DISABLED_CERTIFICATE_MESSAGE);
           }
         },
-        () -> ctx.reportIssue(resource.type(), String.format(MISSING_CERTIFICATE, "negotiateClientCertificate")));
+        () -> ctx.reportIssue(resource.type(), String.format(MISSING_CERTIFICATE_MESSAGE, "negotiateClientCertificate")));
+  }
+
+  private static void checkContainerApps(CheckContext ctx, ResourceDeclaration resource) {
+    if (isResourceVersionBefore(resource, "2022-10-01")) {
```

Comment:
In the ticket, it should be `2022-10-01` or newer. IMO the method name supposed that the opposite is tested.